src/components/flowchart/components/sliced-pipeline-action-bar/sliced-pipeline-action-bar.js

```diff
@@ -1,27 +1,69 @@
 import React from 'react';
+import classnames from 'classnames';
 import Button from '../../../ui/button';
+import CommandCopier from '../../../ui/command-copier/command-copier';
+import CutIcon from '../../../icons/cut';
+import IconButton from '../../../ui/icon-button';
 
 import './sliced-pipeline-action-bar.scss';
 
-export const SlicedPipelineActionBar = ({
-  chartSize,
-  slicedPipeline,
-  visibleSidebar,
-}) => {
-  const { outerWidth } = chartSize;
+export const SlicedPipelineActionBar = React.forwardRef((props, ref) => {
+  const {
+    chartSize,
+    displayMetadataPanel,
+    isSlicingPipelineApplied,
+    onApplySlicingPipeline,
+    onResetSlicingPipeline,
+    runCommand,
+    slicedPipeline,
+    visibleSidebar,
+  } = props;
+  const { outerWidth: screenWidth } = chartSize;
+  const actionBarWidth =
+    ref.current && ref.current.firstChild.getBoundingClientRect().width;
+
+  const metaDataPanelWidth = displayMetadataPanel ? 600 : 400;
+  const nodeListWidth = visibleSidebar ? 200 : 400;
+
+  const transformX =
+    screenWidth - nodeListWidth - metaDataPanelWidth - actionBarWidth / 2;
```

Comment:
good spot, I'm meant to add the minimum value for transformX so it never gets hidden behind the nodelist bar. Updated the code now

Reply:
@Huongg on `demo-slicing-pipeline-final` branch I tested locally I still see action bar overlap with metadata panel.

<img width="1800" alt="Screenshot 2024-08-13 at 2 31 55 p m" src="https://github.com/user-attachments/assets/012fa54a-000a-4591-b3cb-08448f9d272b">

Reply:
@Huongg I am fine with it. It would have been nice to fit action bar within available space, keeping command box as variable width container, may be we can keep this as future work after MVP.

Reply:
yeah interesting idea, i think it require further design thinking as in order to fix the whole action bar in the available space, we will need to reduce the width, in that case what the transition/animation would look like, etc

Agree that can be something to be considered in the future MVP